apps/labs/posts/code-generator-and-nebari.md

```diff
@@ -0,0 +1,159 @@
+
+<center>
+<img src='/posts/code-generator-and-nebari/nebari-image.png' alt='The Nebari-server Option in Code-Generator App' width="250px">
+</img>
+</center>
+
+To explain it quickly, let’s see how the app works when you click on `Open in Nebari` . After the button is clicked, we use a `netlify` function to commit a zip file and Jupyter notebook using `github/octokit` to [PyTorch-Ignite/nbs](https://github.com/pytorch-ignite/nbs) repository. We can see a committed notebook example below. To understand this better, you can read the code in [PyTorch-Ignite/code-generator/functions/nebari.js](https://github.com/pytorch-ignite/code-generator/tree/main/functions/nebari.js).
+
+![GitHub Notebook Pushed By Code-Generator](/posts/code-generator-and-nebari/github-notebook-push.png)
+
+Now, to integrate this with the `Nebari` server , we used an extension called [Jupyterlab-Github](https://github.com/jupyterlab/jupyterlab-github). I wrote a `netlify` function to create a URL that can use this extension, open the generated link in the new tab, pull the notebook committed by the above `netlify` function, and open this notebook in the server. An example can be seen here. Pretty cool, right?
+
+![Template Opened in Nebari](/posts/code-generator-and-nebari/template-open-in-nebari.png)
+
+For this, I appreciate the Jupyter community's extension ecosystem. They have great extensions for everything and maybe in the future, I may work or try to add more extensions to increase the functionality in the app. Also, I would like to thank the Nebari-dev community here, who provided such great support in testing this in the Nebari server.
+
+Some of the related PRs can be found here:[#314](https://github.com/pytorch-ignite/code-generator/pull/314) [#265](https://github.com/pytorch-ignite/code-generator/pull/265)
+
+### 3. The changes in the CI and other Vuejs based parts of the code base (mainly based on JS)
+
+I had to contribute some JavaScript-based code to the app, but since I had not worked on a JS project before, I needed to learn the basics before contributing to these issues, especially [Vue.js](https://vuejs.org) :-( . So, I tried to learn by reading some online tutorials and documentation. I particularly liked this [tutorial](https://www.youtube.com/watch?v=YrxBCBibVo0&list=PL4cUxeGkcC9hYYGbV60Vq3IXYNfDk8At1) series on YouTube.
+
+To explain more about how the generates templates, let's assume I made some files like `main.py`, `model.py`, and `utils.py` as a template. Now we use the [ejs](https://ejs.co) project to design the templates. To illustrate this, let us take an example from the templates.
```

Comment:
Maybe you mean
```suggestion
To explain more about how to generates templates, let's assume I made some files like `main.py`, `model.py`, and `utils.py` as a template. Now we use the [ejs](https://ejs.co) project to design the templates. To illustrate this, let us take an example from the templates.
```

Reply:
This paragraph is a bit confusing, because it seems like you are using "template" to refer to two different types of templates: `main.py`, `model.py` and `utils.py` on one side, and your generated templates on the other. Is there a way to explain this better?